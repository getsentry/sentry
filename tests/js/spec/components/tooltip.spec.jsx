import React from 'react';

import {mount, mountWithTheme} from 'sentry-test/enzyme';

import Tooltip from 'app/components/tooltip';

describe('Tooltip', function() {
  it('renders', function() {
    const wrapper = mountWithTheme(
      <Tooltip title="test">
        <span>My Button</span>
      </Tooltip>
    );
    expect(wrapper).toSnapshot();
    expect(wrapper).toMatchSnapshot();
  });

  it('updates title', function() {
    const wrapper = mountWithTheme(
      <Tooltip title="test">
        <span>My Button</span>
      </Tooltip>,
      TestStubs.routerContext()
    );

    wrapper.setProps({title: 'bar'});
    wrapper.update();
    const trigger = wrapper.find('span');
    trigger.simulate('mouseEnter');

    const tooltip = document.querySelector('#tooltip-portal .tooltip-content');
    // Check the text node.
    expect(tooltip.childNodes[0].nodeValue).toEqual('bar');

    trigger.simulate('mouseLeave');
  });

  it('disables and does not render', function() {
    const wrapper = mount(
      <Tooltip title="test" disabled>
        <span>My Button</span>
      </Tooltip>,
      TestStubs.routerContext()
    );
    const trigger = wrapper.find('span');
    trigger.simulate('mouseEnter');

    const tooltip = document.querySelector('#tooltip-portal .tooltip-content');
    expect(tooltip).toBeFalsy();

    trigger.simulate('mouseLeave');
  });

  it('does not render an empty tooltip', function() {
    const wrapper = mountWithTheme(
      <Tooltip title="">
        <span>My Button</span>
      </Tooltip>,
      TestStubs.routerContext()
    );
    const trigger = wrapper.find('span');
    trigger.simulate('mouseEnter');

    const tooltipContent = wrapper.find('TooltipContent');
    expect(tooltipContent.prop('hide')).toBe(true);

    trigger.simulate('mouseLeave');
  });
});
