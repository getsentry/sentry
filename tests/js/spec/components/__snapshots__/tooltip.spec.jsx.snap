// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip renders 1`] = `
<Tooltip
  containerDisplayMode="inline-block"
  position="top"
  title="test"
>
  <Manager>
    <Reference>
      <InnerReference
        setReferenceNode={[Function]}
      >
        <span
          aria-describedby="tooltip-123456"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          My Button
        </span>
      </InnerReference>
    </Reference>
  </Manager>
</Tooltip>
`;
